### PR TITLE
Add TLS support in FRPS and FRPC binary

### PR DESCRIPTION
**Pull request by abidlabs**

### CODE REVIEW

File: Makefile.cross-compiles [hunk 1]
```diff
@@ -19,7 +19,15 @@ app:
 		env CGO_ENABLED=0 GOOS=$${os} GOARCH=$${arch} GOMIPS=$${gomips} go build -trimpath -ldflags "$(LDFLAGS)" -o ./release/frps_$${target_suffix} ./cmd/frps;\
 		echo "Build $${os}-$${arch} done";\
 	)
-	@mv ./release/frpc_windows_386 ./release/frpc_windows_386.exe
-	@mv ./release/frps_windows_386 ./release/frps_windows_386.exe
-	@mv ./release/frpc_windows_amd64 ./release/frpc_windows_amd64.exe
-	@mv ./release/frps_windows_amd64 ./release/frps_windows_amd64.exe
+	@if [ -f ./release/frpc_windows_386 ]; then \
```
Comment (abidlabs) [inline Makefile.cross-compiles:22]: Very minor change that allows excluding the windows binary (useful while testing)

File: cmd/frpc/sub/root.go [hunk 2]
```diff
@@ -74,6 +74,7 @@ var (
 	bindPort          int
 
 	tlsEnable bool
+	tls_trusted_ca_file string
```
Comment (abidlabs) [inline cmd/frpc/sub/root.go]: Adds a parameter `tls_trusted_ca_file` to the FPRC binary that allows passing in a path to a root certificate file

File: scripts/frps_tls.ini [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+log_level = trace
+tcp_mux = true
+bind_port = 7000
+dashboard_port = 7001
+vhost_http_port = 80
+subdomain_host = gradio.live
+detailed_errors_to_client = false
+custom_404_page = /etc/frp/404.html
+
+tls_enable = true
```
Comment (abidlabs) [inline scripts/frps_tls.ini:11]: Updates the FRPS configuration file to include private key and certificate files (these should remain private)

**Review by aliabd — APPROVED** [review]

**Comment by abidlabs** [comment]
Thanks @aliabd!

**Comment by abidlabs** [comment]
cc @XciD for visiblity

**Pull request merged**